backend/src/lib/knex/scim.ts

```diff
@@ -75,42 +74,34 @@ export const generateKnexQueryFromScim = (
         break;
       }
       case "and": {
-        void query.andWhere((subQueryBuilder) => {
-          scimFilterAst.filters.forEach((el) => {
-            stack.push({
-              query: subQueryBuilder,
-              scimFilterAst: el
-            });
+        scimFilterAst.filters.forEach((el) => {
```

Comment:
So basically this is needed when the scim query does operation based on a scim specific query language.
Like `Get /Users?eq=username something and `

Reply:
oh i see and so if the query has a `and` we want to re scan the full query against the `processDynamicQuery` recursively?

Reply:
Ya because that's how knex js work. 

The and is a scope and to make that scope in knexjs it needs to be a function recursively. I tried stack based but due to the callback way of knexjs pure stack based iterative is not possible.